src/net/parse.cpp

```diff
@@ -34,28 +34,67 @@
 
 namespace net
 {
+    void get_network_address_host_and_port(const std::string& address, std::string& host, std::string& port)
+    {
+        // require ipv6 address format "[addr:addr:addr:...:addr]:port"
+        if (address.find(']') != std::string::npos)
+        {
+            host = address.substr(1, address.rfind(']') - 1);
+            if ((host.size() + 2) < address.size())
+            {
+                port = address.substr(address.rfind(':') + 1);
+            }
+        }
+        else
+        {
+            host = address.substr(0, address.rfind(':'));
+            if (host.size() < address.size())
+            {
+                port = address.substr(host.size() + 1);
+            }
+        }
+    }
```

Comment:
That seems way underspecified.

Reply:
Agreed that it seems that way, but it's just a helper function to avoid a bit of code duplication; the places it is called will handle making sure what it spits out is correct/usable.